src/shared/bg_pmove.cpp

```diff
@@ -4865,8 +4852,7 @@ static bool  PM_SlideMove( bool gravity )
 
 		// see if we can make it there
 		// spectators ignore movers, so that they can noclip through doors
-		pm->trace( &trace, pm->ps->origin, pm->mins, pm->maxs, end, pm->ps->clientNum,
-		           pm->tracemask, ( pm->ps->pm_type == PM_SPECTATOR ) ? CONTENTS_MOVER : 0 );
+		pm->trace( &trace, pm->ps->origin, pm->mins, pm->maxs, end, pm->ps->clientNum, pm->tracemask, ( pm->ps->pm_type == PM_SPECTATOR ) ? CONTENTS_MOVER : 0 );
```

Comment:
Globally its a good idea, in practice I try to even stay under 80, but in *this* case (as in, this PR, the state of this file, etc), it's useful to move the stuff on a single line to more easily identify blocks of duplicated stuff.
Still considering *this* specific case, I suspect more specialized APIs could be done which would reduce the line's width and increase clarity.

Globally, I think trap_Trace's API:

* only hides a useful parameter (the type of trace)
* could *return* the trace instead of using it as parameter
* end position and start position are both mandatory, end position should be right after start
* most cases if not all use either raw `start`, `end`, `mins` and `maxs`, which belongs to `clientNum`  OR they do a VectorMA on both `start` and `end`, which is itself more than often just a displaced `start`

I am almost certain most of the trap_Trace (even if many will be moved to actually calling CM_Trace_world or something like that in some of my later commits, here or in another PR I don't remember, so that they hide less info and certainly stop misleading user in thinking that it's a syscall) can be fused with around them into a proper function.

So, again, yes, usually it's better to avoid long lines, but breaking lines, especially in not obvious places, makes things worst.
In any cases, I would have changed the line break here, if keeping the line small then I would have split like this:

```
		pm->trace( &trace, pm->ps->origin, pm->mins, pm->maxs, end,
		           pm->ps->clientNum, pm->tracemask, ( pm->ps->pm_type == PM_SPECTATOR ) ? CONTENTS_MOVER : 0 );
```

So that on 1st line, you (almost) only get the 3D stuff (and the dumb trace pointer for 0 reason) and the other data on 2nd line. This would result in 2nd line *still* being longer than 100, which is still annoying but more logical.
I think line limitations should be a recommendation, not a strong, hard rule.

Reply:
The type of trace is not a useful parameter as long as our game is entirely based on bounding *boxes*.

Reply:
Then use a default value, instead of a proxy function which only makes things more complicated. IIRC I have something which does just that: replace the trap_Trace things with the real one later, in most cases. Most cases because the shared code uses callbacks, and I'm not sure I've updated those accordingly.